### PR TITLE
PSTRESS-45 Pstress crashes if log directory does not exist

### PR DESCRIPTION
**Pull request by mohitj1988**

### CODE REVIEW

File: src/node.cpp [hunk 1]
```diff
@@ -19,9 +19,20 @@ void Node::end_node() {
     delete querylist;
 }
 
+/* create the logdir if does not exist */
+void create_log_dir() {
```
Comment (satya-bodapati) [inline src/node.cpp:23]: have a bool and return false on error

File: src/node.cpp [hunk 2]
```diff
@@ -19,9 +19,20 @@ void Node::end_node() {
...
+  if(stat(logdir_path.c_str(),&st) != 0) {
+    if (mkdir(logdir_path.c_str(), 0755) == -1)
+      std::cerr << "Could not create log dir:  " << strerror(errno) << std::endl;
+  }
+}
+
 bool Node::createGeneralLog() {
   std::string logName;
   logName = myParams.logdir + "/" + myParams.myName + "_general" + ".log";
+  create_log_dir();
```
Comment (satya-bodapati) [inline src/node.cpp:35]: return error to the callers and up and ensure it exits gracefully

File: src/node.hpp [hunk 3]
```diff
@@ -11,6 +11,10 @@
 #include <string>
 #include <thread>
 #include <vector>
+#include <sys/stat.h>
+
+/* Function prototype */
```
Comment (satya-bodapati) [inline src/node.hpp:16]: not so useful comment.
/* Create log dir if not exists */

Comment (satya-bodapati) [reply src/node.hpp:16]: may be even avoid this. and create a static function. I assume it is not used outside this file..

File: src/node.cpp [hunk 4]
```diff
@@ -20,19 +20,25 @@ void Node::end_node() {
...
 
 bool Node::createGeneralLog() {
   std::string logName;
   logName = myParams.logdir + "/" + myParams.myName + "_general" + ".log";
-  create_log_dir();
+  bool is_success = create_log_dir();
+  if (!is_success) {
+    std::cerr << "Could not create log dir: " << strerror(errno) << std::endl;
+    exit(1);
+  }
```
Comment (satya-bodapati) [inline src/node.cpp:41]: Can we just return an error the caller and it should recursively send to first caller and it should exit.

File: src/node.cpp [hunk 5]
```diff
@@ -20,19 +20,25 @@ void Node::end_node() {
 }
 
 /* create the logdir if does not exist */
```
Comment (satya-bodapati) [inline src/node.cpp:22]:
```suggestion
/* create the logdir if does not exist 
@return false on error, true on success */
```

File: src/node.cpp [hunk 6]
```diff
@@ -20,19 +20,25 @@ void Node::end_node() {
...
-void create_log_dir() {
+bool create_log_dir() {
```
Comment (satya-bodapati) [inline src/node.cpp:23]:
```suggestion
static bool create_log_dir() {
```

File: src/node.hpp [hunk 7]
```diff
@@ -13,8 +13,8 @@
 #include <vector>
 #include <sys/stat.h>
 
-/* Function prototype */
-void create_log_dir();
+/* Create log dir if not exists */
+bool create_log_dir();
```
Comment (satya-bodapati) [inline src/node.hpp:17]: get rid of this, we will now use static function

File: src/node.cpp [hunk 8]
```diff
@@ -20,19 +20,25 @@ void Node::end_node() {
...
   std::string logdir_path = opt_string(LOGDIR);
```
Comment (satya-bodapati) [inline src/node.cpp:24]:
```suggestion
  std::string logdir_path(opt_string(LOGDIR));
```

File: src/node.hpp (hunk 3 above)
Comment (mohitj1988) [reply src/node.hpp]: Done

File: src/node.cpp (hunk 4 above)
Comment (mohitj1988) [reply src/node.cpp]: Done

File: src/node.cpp (hunk 5 above)
Comment (mohitj1988) [reply src/node.cpp]: Done

File: src/node.cpp (hunk 6 above)
Comment (mohitj1988) [reply src/node.cpp]: Done

File: src/node.hpp (hunk 7 above)
Comment (mohitj1988) [reply src/node.hpp]: Done

File: src/node.cpp (hunk 8 above)
Comment (mohitj1988) [reply src/node.cpp:26]: With the suggested changes, there was a compilation error. So, I retained the same.

**Pull request merged**